src/main/kotlin/xyz/atrius/waystones/commands/GetKeyCommand.kt

```diff
@@ -0,0 +1,39 @@
+package xyz.atrius.waystones.commands
+
+import org.bukkit.Bukkit
+import org.bukkit.command.CommandSender
+import org.bukkit.entity.Player
+import xyz.atrius.waystones.utility.defaultWarpKey
+import xyz.atrius.waystones.utility.pluralize
+import xyz.atrius.waystones.utility.translateColors
+
+object GetKeyCommand : SimpleCommand("getkey", "gk", "key") {
+    // Calculate Player and Amount Values
+    private fun getPlayerAndAmount(default: Player, first: String, second: String): Pair<Player, Int> {
+        val player = Bukkit.getPlayer(first) ?: default
+        val amount = second.toIntOrNull() ?: first.toIntOrNull() ?: 1
+        return player to amount
+    }
+
+    override fun execute(sender: CommandSender, args: Array<String>) {
+        // Set Amount and Player to give WarpKeys to
+        val (player, amount) = getPlayerAndAmount(
+            sender as Player,
+            args.getOrNull(0) ?: "",
+            args.getOrNull(1) ?: ""
+        )
+        // Check Permissions
+        if (
+            !sender.hasPermission("waystones.getkey.self") || // Give Key to Self
+            (!sender.hasPermission("waystones.getkey.all") && sender != player)
+        ) { // Give Key to Other
```

Comment:
Perhaps we should format it like this: 
```kt
		// Give Key to other players
        if (!sender.hasPermission("waystones.getkey.self") 
        	|| !sender.hasPermission("waystones.getkey.all") 
	        && sender != player
        ) {
```

Reply:
i feel this wouldn't function as intended. two criteria are being checked here and if i'm not mistaken the extra pair of parenthesis are important to that check

i won't deny the formatting is difficult to read as is, but this seems equally as hard and function in unintended ways

Reply:
The extra set of parentheses was deemed unnecessary by intelliJ. Given that the operations are checked sequentially and the order given this sorta makes sense as to why.

Reply:
seems i was mistaken about how kotlin handles this, this does indeed work as intended, i've pushed the change